src/bootstrap/native.rs

```diff
@@ -133,86 +133,382 @@ fn check_llvm_version(build: &Build, llvm_config: &Path) {
+            "arm/umodsi3.S",
+        ]);
+    }
+
+    if target.contains("armv7") {
+        sources.extend(vec![
+            "arm/sync_fetch_and_add_4.S",
+            "arm/sync_fetch_and_add_8.S",
+            "arm/sync_fetch_and_and_4.S",
+            "arm/sync_fetch_and_and_8.S",
+            "arm/sync_fetch_and_max_4.S",
+            "arm/sync_fetch_and_max_8.S",
+            "arm/sync_fetch_and_min_4.S",
+            "arm/sync_fetch_and_min_8.S",
+            "arm/sync_fetch_and_nand_4.S",
+            "arm/sync_fetch_and_nand_8.S",
+            "arm/sync_fetch_and_or_4.S",
+            "arm/sync_fetch_and_or_8.S",
+            "arm/sync_fetch_and_sub_4.S",
+            "arm/sync_fetch_and_sub_8.S",
+            "arm/sync_fetch_and_umax_4.S",
+            "arm/sync_fetch_and_umax_8.S",
+            "arm/sync_fetch_and_umin_4.S",
+            "arm/sync_fetch_and_umin_8.S",
+            "arm/sync_fetch_and_xor_4.S",
+            "arm/sync_fetch_and_xor_8.S",
+        ]);
+    }
+
+    if target.contains("eabihf") {
+        sources.extend(vec![
+            "arm/adddf3vfp.S",
+            "arm/addsf3vfp.S",
+            "arm/divdf3vfp.S",
+            "arm/divsf3vfp.S",
+            "arm/eqdf2vfp.S",
+            "arm/eqsf2vfp.S",
+            "arm/extendsfdf2vfp.S",
+            "arm/fixdfsivfp.S",
+            "arm/fixsfsivfp.S",
+            "arm/fixunsdfsivfp.S",
+            "arm/fixunssfsivfp.S",
+            "arm/floatsidfvfp.S",
+            "arm/floatsisfvfp.S",
+            "arm/floatunssidfvfp.S",
+            "arm/floatunssisfvfp.S",
+            "arm/gedf2vfp.S",
+            "arm/gesf2vfp.S",
+            "arm/gtdf2vfp.S",
+            "arm/gtsf2vfp.S",
+            "arm/ledf2vfp.S",
+            "arm/lesf2vfp.S",
+            "arm/ltdf2vfp.S",
+            "arm/ltsf2vfp.S",
+            "arm/muldf3vfp.S",
+            "arm/mulsf3vfp.S",
+            "arm/negdf2vfp.S",
+            "arm/negsf2vfp.S",
+            "arm/nedf2vfp.S",
+            "arm/nesf2vfp.S",
+            "arm/restore_vfp_d8_d15_regs.S",
+            "arm/save_vfp_d8_d15_regs.S",
+            "arm/subdf3vfp.S",
+            "arm/subsf3vfp.S",
+            "arm/truncdfsf2vfp.S",
+            "arm/unorddf2vfp.S",
+            "arm/unordsf2vfp.S",
+        ]);
+    }
+
+    if target.contains("aarch64") {
+        sources.extend(vec![
+            "comparetf2.c",
+            "extenddftf2.c",
+            "extendsftf2.c",
+            "fixtfdi.c",
+            "fixtfsi.c",
+            "fixtfti.c",
+            "fixunstfdi.c",
+            "fixunstfsi.c",
+            "fixunstfti.c",
+            "floatditf.c",
+            "floatsitf.c",
+            "floatunditf.c",
+            "floatunsitf.c",
+            "multc3.c",
+            "trunctfdf2.c",
+            "trunctfsf2.c",
+        ]);
+    }
+
+    let mut out_of_date = false;
+    for src in sources {
+        let src = build.src.join("src/compiler-rt/lib/builtins").join(source);
```

Comment:
`rustc` will fail to compile here. That argument to `join` there should be `src`, right?